src/design/nova-decider-offchain.md

```diff
@@ -35,36 +38,46 @@ In the offchain case, since we can end up with proofs in both curves of the cycl
 
 #### Circuit1 $\in Fr$ ($E_1.F_r$)
 
-- 1.1: check that the given NIFS challenge $r$ is indeed well computed. This challenge is then used outside the circuit by the Verifier to compute NIFS.V obtaining $U_{i+1}$
-- 2: check that $u_n.\overline{E}=0$ and $u_n.u=1$
-- 3: check that $u_n.x_0 = H(n, z_0, z_n, U_n)$ and $u_n.x_1 = H(U_{EC,n})$
-- 4: correct RelaxedR1CS relation of $U_{n+1}, W_{n+1}$ of the AugmentedFCircuit
-- 5.1: Check correct computation of the CommitmentScheme challenges for $U_{n+1}$
-    $$c_E = H(U_{n+1}.\overline{E}.\{x,y\}),~~c_W = H(U_{n+1}.\overline{W}.\{x,y\})$$
+- 1: Enforce $U_{n+1}$ and $W_{n+1}$ satisfy $r1cs$, the RelaxedR1CS relation of the AugmentedFCircuit
+- 2: Check that $u_n.\overline{E}=0$ and $u_n.u=1$
+- 3: Check that $u_n.x_0 = H(n, z_0, z_n, U_n)$ and $u_n.x_1 = H(U_{EC,n})$
+- 6.1: Partially enforce that $U_{n + 1}$ is the correct folding of $U_n, u_n$
+    - By partially, we mean that only field elements in $U_{n + 1}$ are checked, while the group elements (i.e., commitments) are not, as they are in $E_1$ and are expensive non-native operations.
+- 7.1: Check correct computation of the KZG challenges
+    $$c_E = H(\overline{E}.\{x,y\}),~~c_W = H(\overline{W}.\{x,y\})$$
     which we do through in-circuit Transcript.
-- 5.2: check that the CommitmentScheme evaluations for $U_{n+1}$ are correct
+- 7.2: Check that the KZG evaluations are correct
```

Comment:
(the current implementation uses KZG, but I dream of a day where we can just place it by other schemes like Pedersen)

Reply:
Opened [an issue](https://github.com/privacy-scaling-explorations/sonobe/issues/171) for Pedersen support :)